client/matching/client.go

```diff
@@ -77,18 +86,16 @@ func (c *clientImpl) AddActivityTask(
 	ctx context.Context,
 	request *matchingservice.AddActivityTaskRequest,
 	opts ...grpc.CallOption) (*matchingservice.AddActivityTaskResponse, error) {
-	partition := c.loadBalancer.PickWritePartition(
-		namespace.ID(request.GetNamespaceId()),
+	partition, err := c.pickPartitionForWrite(
```

Comment:
These functions AddActivity, AddWorkflow, etc. all look the same with a little variance. You can encapsulate differences and pass them into one common function.